EasyPost.Tests/ServicesTests/Beta/RateServiceTest.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using EasyPost.Calculation;
+using EasyPost.Models.API.Beta;
+using EasyPost.Tests._Utilities;
+using EasyPost.Tests._Utilities.Annotations;
+using EasyPost.Utilities.Annotations;
+using Xunit;
+
+namespace EasyPost.Tests.ServicesTests.Beta
+{
+    public class RateServiceTests : UnitTest
+    {
+        public RateServiceTests() : base("beta_rate_service")
+        {
+        }
+
+        #region Tests
+
+        #region Test CRUD Operations
+
+        [Fact]
+        [CrudOperations.Read]
+        [Testing.Function]
+        public async Task TestRetrieve()
+        {
+            UseVCR("retrieve");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            Assert.IsType<List<StatelessRate>>(rates);
+        }
+
+        #endregion
+
+        [Fact]
+        [Testing.Function]
+        public async Task TestGetLowestRateInstanceFunction()
+        {
+            UseVCR("get_lowest_rate_instance_function");
+
+            Dictionary<string, object> data = Fixtures.BasicShipment;
+
+            List<StatelessRate> rates = await Client.Beta.Rate.RetrieveStatelessRates(data);
+
+            StatelessRate lowestStatelessRate = Calculation.Rates.GetLowestStatelessRate(rates);
```

Comment:
hmm, I'd expect user-facing functions to be in `Utils`, not `Calculation`.

Reply:
Calculation is public, but realistically was not meant for end-user consumption. Users are expected to access rate filtering via the Rate service (even though it doesn't make an API call).

If we want to preserve "services" for solely API call functions and move all non-API call functions into a set of publicly-accessible utilities, that's a design choice we should discuss.

Reply:
We did, it's just C# is now behind. Java and PHP along with soon-to-be Node have their non-API functions in the Utils namespace.

Reply:
Should we preserve the "old" way and fix this all at once in the future, or start splitting it now?

Reply:
Oh, my suggestion is that since this is a fresh function, we set it up the right way, how we want things to be, and in the next major release we can swap things we don't love to match.

Reply:
Alright, the legacy methods have been marked as deprecated and redirect to the functions now located in "Utilities"